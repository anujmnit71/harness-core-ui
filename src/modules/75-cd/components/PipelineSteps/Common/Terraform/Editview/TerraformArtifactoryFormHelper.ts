/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */
import * as Yup from 'yup'
import { TerraformStoreTypes } from '../TerraformInterfaces'

export const formatInitialValues = (isConfig: boolean, prevStepData: any, isTerraformPlan: boolean) => {
  if (isConfig && isTerraformPlan) {
    return {
      spec: {
        configuration: {
          configFiles: {
            store: {
              spec: {
                repositoryName:
                  prevStepData.formValues?.spec?.configuration?.configFiles?.store.spec?.repositoryName || null,
                artifacts: prevStepData.formValues?.spec?.configuration?.configFiles?.store.spec?.artifacts || []
              }
            }
          }
        }
      }
    }
  }

  if (isConfig) {
    return {
      spec: {
        configuration: {
          configFiles: {
            store: {
              spec: {
                repositoryName:
                  prevStepData.formValues?.spec?.configuration?.spec?.configFiles?.store.spec?.repositoryName || null,
                artifacts: prevStepData.formValues?.spec?.configuration?.spec?.configFiles?.store.spec?.artifacts || []
              }
            }
          }
        }
      }
    }
  }

  return {
    varFile: {
      identifier: prevStepData?.varFile?.identifier || null,
      type: TerraformStoreTypes.Remote,
      spec: {
        store: {
          spec: {
            repositoryName: prevStepData?.varFile?.spec?.store?.spec?.repositoryName || null,
            artifacts: prevStepData?.varFile?.spec?.store?.spec?.artifacts || []
          }
        }
      }
    }
  }
}

export const getConnectorRef = (isConfig: boolean, isTerraformPlan: boolean, prevStepData: any) => {
  if (isConfig && isTerraformPlan) {
    return prevStepData?.formValues.spec?.configuration?.configFiles?.store?.spec?.connectorRef
  }

  if (isConfig) {
    return prevStepData?.formValues?.spec?.configuration?.spec?.configFiles?.store?.spec?.connectorRef
  }

  return prevStepData?.varFile?.spec?.store?.spec?.connectorRef
}

export const terraformArtifactorySchema = (isConfig: boolean, getString: any) => {
  const artifacts = {
    repositoryName: Yup.string().required(getString('common.validation.identifierIsRequired')),
    artifacts: Yup.array().of(
      Yup.object().shape({
        artifactFile: Yup.object().shape({
          artifactPathExpression: Yup.string().required(getString('cd.pathCannotBeEmpty')),
          name: Yup.string().required(getString('cd.pathCannotBeEmpty'))
        })
      })
    )
  }

  const configSetup = {
    configFiles: Yup.object().shape({
      store: Yup.object().shape({
        spec: Yup.object().shape({
          ...artifacts
        })
      })
    })
  }

  if (isConfig) {
    return Yup.object().shape({
      spec: Yup.object().shape({
        configuration: Yup.object().shape({
          ...configSetup
        })
      })
    })
  }

  return {
    varFile: Yup.object().shape({
      identifier: Yup.string().required(getString('common.validation.identifierIsRequired')),
      spec: Yup.object().shape({
        store: Yup.object().shape({
          spec: Yup.object().shape({
            ...artifacts
          })
        })
      })
    })
  }
}

export const tfArtifactoryFormINputNames = (isConfig: boolean) => {
  if (isConfig) {
    return {
      repositoryName: 'spec.configuration.configFiles.store.spec.repositoryName',
      artifacts: 'spec.configuration.configFiles.store.spec.artifacts'
    }
  }

  return {
    repositoryName: 'varFile.spec.store.spec.repositoryName',
    artifacts: 'varFile.spec.store.spec.artifacts'
  }
}

export const formatOnSubmitData = (values: any, prevStepData: any, connectorValue: any) => {
  const payload = {
    ...values,
    connectorRef: connectorValue
  }

  const data = {
    varFile: {
      type: payload.varFile.type,
      identifier: payload.varFile.identifier,
      spec: {
        store: {
          type: payload.connectorRef?.connector?.type || prevStepData?.selectedType,
          spec: {
            ...payload.varFile.spec?.store?.spec,
            connectorRef: payload.connectorRef
          }
        }
      }
    }
  }
  return data
}
