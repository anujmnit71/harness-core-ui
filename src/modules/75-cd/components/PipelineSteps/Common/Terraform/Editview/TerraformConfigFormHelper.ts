/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { Connectors } from '@connectors/constants'
import type { ConnectorInfoDTO } from 'services/cd-ng'
import type { StringKeys } from 'framework/strings'

export const AllowedTypes = ['Git', 'Github', 'GitLab', 'Bitbucket']
export type ConnectorTypes = 'Git' | 'Github' | 'GitLab' | 'Bitbucket'

export const tfVarIcons: any = {
  Git: 'service-github',
  Github: 'github',
  GitLab: 'service-gotlab',
  Bitbucket: 'bitbucket'
}

export const ConnectorMap: Record<string, ConnectorInfoDTO['type']> = {
  Git: Connectors.GIT,
  Github: Connectors.GITHUB,
  GitLab: Connectors.GITLAB,
  Bitbucket: Connectors.BITBUCKET
}

export const ConnectorLabelMap: Record<ConnectorTypes, StringKeys> = {
  Git: 'pipeline.manifestType.gitConnectorLabel',
  Github: 'common.repo_provider.githubLabel',
  GitLab: 'common.repo_provider.gitlabLabel',
  Bitbucket: 'pipeline.manifestType.bitBucketLabel'
}
