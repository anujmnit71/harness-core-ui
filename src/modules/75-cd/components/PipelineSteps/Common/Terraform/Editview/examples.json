{
  "artifacts": [
    {
      "artifactName": "config.tf.zip",
      "artifactPath": "generic-local/terraform/config.tf.zip"
    },
    {
      "artifactName": "localresource.tfvar.zip",
      "artifactPath": "generic-local/terraform/localresource.tfvar.zip"
    },
    {
      "artifactName": "localresourcemodule.zip",
      "artifactPath": "generic-local/terraform/localresourcemodule.zip"
    },
    {
      "artifactName": "lovalresourcetfvarinnerfolder.zip",
      "artifactPath": "generic-local/terraform/lovalresourcetfvarinnerfolder.zip"
    }
  ],
  "repositories": {
    "harness-maven-snapshots-v2": "harness-maven-snapshots-v2",
    "thirdparty-annonymous": "thirdparty-annonymous",
    "helm": "helm",
    "todolist-zip": "todolist-zip",
    "FloraRepo-generic": "FloraRepo-generic",
    "se-demo": "se-demo",
    "KenHarnessHelm": "KenHarnessHelm",
    "harness-maven": "harness-maven",
    "lambda": "lambda",
    "venkat-generic": "venkat-generic",
    "workshop-local": "workshop-local",
    "iis-website": "iis-website",
    "sales-test": "sales-test",
    "alex-test2": "alex-test2",
    "progresso": "progresso",
    "usheer-repo": "usheer-repo",
    "workshop": "workshop",
    "todolist": "todolist",
    "generic-local": "generic-local",
    "anshul-local": "anshul-local",
    "kentest1": "kentest1",
    "functional-test": "functional-test",
    "datacollection-dsl": "datacollection-dsl",
    "harness-internal": "harness-internal",
    "harness-cf": "harness-cf",
    "harness-maven-snapshots": "harness-maven-snapshots",
    "harness-bower": "harness-bower",
    "generic-repo": "generic-repo",
    "iistest": "iistest",
    "cf-tutorial": "cf-tutorial",
    "TestSkyHighMaven": "TestSkyHighMaven",
    "Anonymous": "Anonymous",
    "harness-release": "harness-release",
    "ravi-generic": "ravi-generic",
    "helm-remote": "helm-remote",
    "harness-public": "harness-public",
    "ivy-local": "ivy-local",
    "micah-helm-charts": "micah-helm-charts",
    "helm-local": "helm-local",
    "BuildsTools": "BuildsTools",
    "citilab-helm-icg-prod": "citilab-helm-icg-prod",
    "Venkat-Test-BT-274": "Venkat-Test-BT-274",
    "KenTest": "KenTest",
    "harness-rpm": "harness-rpm",
    "todolist-tutorial": "todolist-tutorial",
    "harness-helm-charts": "harness-helm-charts",
    "citilab-helm-icg-dev": "citilab-helm-icg-dev",
    "alex-helm-test-repo": "alex-helm-test-repo",
    "ivy-test": "ivy-test",
    "qa-test": "qa-test"
  }
}
