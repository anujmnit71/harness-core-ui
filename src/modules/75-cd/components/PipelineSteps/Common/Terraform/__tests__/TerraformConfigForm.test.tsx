/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import { MultiTypeInputType } from '@wings-software/uicore'
import { TestWrapper } from '@common/utils/testUtils'

import { TerraformConfigStepOne, TerraformConfigStepTwo } from '../Editview/TerraformConfigForm'

jest.mock('services/cd-ng', () => ({
  useGetConnector: jest.fn().mockImplementation(() => ({ data: {} }))
}))

const allowableTypes = [MultiTypeInputType.FIXED, MultiTypeInputType.RUNTIME, MultiTypeInputType.EXPRESSION]
const renderStepOneComponent = (data: any): void => {
  render(
    <TestWrapper>
      <TerraformConfigStepOne
        data={data}
        isReadonly={false}
        isEditMode={false}
        allowableTypes={allowableTypes}
        setConnectorView={jest.fn()}
        setSelectedConnector={jest.fn()}
        selectedConnector={'Git'}
      />
    </TestWrapper>
  )
}

const testData = {
  spec: {
    configuration: {
      configFiles: {
        store: {
          type: 'Git',
          spec: {
            skipResourceVersioning: false,
            store: {
              type: 'Git',
              spec: {
                connectorRef: 'Git',
                gitFetchType: 'Branch',
                branch: 'master',
                repoName: 'repoName',
                commidId: undefined
              }
            }
          }
        }
      }
    }
  }
}

describe('TerraformConfigForm StepOne tests', () => {
  test(`renders without crashing with no initial data`, async () => {
    renderStepOneComponent({})
    // all connector options displayed
    const gitConnector = await screen.findByTestId('varStore-Git')
    expect(gitConnector).toBeInTheDocument()

    const gitlabConnector = await screen.findByTestId('varStore-GitLab')
    expect(gitlabConnector).toBeInTheDocument()

    const githubbConnector = await screen.findByTestId('varStore-Github')
    expect(githubbConnector).toBeInTheDocument()

    const bitBucketConnector = await screen.findByTestId('varStore-Bitbucket')
    expect(bitBucketConnector).toBeInTheDocument()
  })

  test(`new connector view works correctly`, async () => {
    renderStepOneComponent({})
    const gitConnector = await screen.findByTestId('varStore-Git')
    fireEvent.click(gitConnector)

    const newConnectorLabel = await screen.findByText('newLabel pipeline.manifestType.gitConnectorLabel connector')
    expect(newConnectorLabel).toBeInTheDocument()
    fireEvent.click(newConnectorLabel)

    const nextStepButton = await screen.findByText('continue')
    expect(nextStepButton).toBeDefined()
    fireEvent.click(nextStepButton)

    expect(screen).toMatchSnapshot()
  })

  test(`new connector view works correctly with previously selected connector`, () => {
    renderStepOneComponent(testData)

    expect(screen).toMatchSnapshot()
  })
})

const renderStepTwoComponent = (data?: any): void => {
  render(
    <TestWrapper>
      <TerraformConfigStepTwo
        isTerraformPlan
        prevStepData={data}
        isReadonly={false}
        allowableTypes={allowableTypes}
        onSubmitCallBack={jest.fn()}
      />
    </TestWrapper>
  )
}

describe('TerraformConfigForm StepTwo tests', () => {
  test(`renders without crashing with no initial data`, async () => {
    renderStepTwoComponent()
    // all inputs are displayed
    const fetchType = await screen.findByPlaceholderText('- pipeline.manifestType.gitFetchTypeLabel -')
    expect(fetchType).toBeInTheDocument()

    const folderPath = await screen.findByPlaceholderText('pipeline.manifestType.pathPlaceholder')
    expect(folderPath).toBeInTheDocument()
  })

  test(`loads data in edit mode`, async () => {
    const prevStepData = {
      spec: {
        configuration: {
          configFiles: {
            store: {
              spec: {
                repoName: 'git name',
                gitFetchType: 'pipelineSteps.commitIdValue',
                commitId: 'test-commit',
                folderPath: 'test-folder',
                connectorRef: {
                  connector: {
                    spec: {
                      connectionType: 'Account'
                    }
                  }
                }
              }
            }
          }
        }
      }
    }

    renderStepTwoComponent(prevStepData)
    const testRepoName = await screen.getByDisplayValue('git name')
    expect(testRepoName).toBeInTheDocument()

    const testFetchType = await screen.getByDisplayValue('gitFetchTypes.fromCommit')
    expect(testFetchType).toBeInTheDocument()

    const testCommitId = await screen.getByDisplayValue('test-commit')
    expect(testCommitId).toBeInTheDocument()

    const testFolderPath = await screen.getByDisplayValue('test-folder')
    expect(testFolderPath).toBeInTheDocument()
  })
})
