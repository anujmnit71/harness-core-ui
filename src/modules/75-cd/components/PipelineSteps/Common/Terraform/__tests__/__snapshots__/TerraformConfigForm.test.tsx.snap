// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TerraformConfigForm StepOne tests new connector view works correctly 1`] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`TerraformConfigForm StepOne tests new connector view works correctly with previously selected connector 1`] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`TerraformConfigForm StepOne tests renders when terraform plan 1`] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`TerraformConfigForm StepTwo tests can press previous button 1`] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`TerraformConfigForm StepTwo tests can update input fields 1`] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`TerraformConfigForm StepTwo tests can update input fields 2`] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`TerraformConfigForm StepTwo tests can update input fields 3`] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`TerraformConfigForm StepTwo tests can update select input field 1`] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`TerraformConfigForm StepTwo tests renders without crashing with no initial data 1`] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`TerraformConfigForm StepTwo tests should not show errors when submitting 1`] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`TerraformConfigForm StepTwo tests should show error message when missing fields 1`] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;
